src/models/OAuthModel.php

```diff
@@ -425,28 +423,59 @@ public function createUserFromTwitterUsername($clientId, $values)
      *
      * @param  string $clientId         aka consumer_key (of the joindin client)
      * @param  string $email            User's email address (that we just got back from authenticating them)
+     * @param  string $fullName         User's full name from Facebook
+     * @param  string $userName         Username to be created if not found
      * @return array|false              Array of access token and user uri on success or false or failure
      */
-    public function createAccessTokenFromTrustedEmail($clientId, $email)
+    public function createAccessTokenFromTrustedEmail($clientId, $email, $fullName = '', $userName = '')
     {
-        $sql = "select ID from user "
-            . "where email = :email "
-            . "and verified = 1";
+        $sql = "
+            SELECT ID from user
+            WHERE email = :email
+            AND verified = 1";
 
         $stmt = $this->_db->prepare($sql);
         $stmt->execute(array("email" => $email));
         $result = $stmt->fetch(PDO::FETCH_ASSOC);
-        if ($result) {
-            $userId = $result['ID'];
+        if (!$result  && $fullName && $userName) {
+            $result = $this->createUserFromTrustedEmail($email, $fullName, $userName);
+        }
 
-            // create new token
-            $accessToken = $this->newAccessToken($clientId, $userId);
+        $userId = $result['ID'];
+        if (!$userId) {
+            return false;
+        }
 
-            // we also want to send back the logged in user's uri
-            $userUri = $this->getUserUri($userId);
+        $accessToken = $this->createAccessToken($clientId, $userId);
 
-            return array('access_token' => $accessToken, 'user_uri' => $userUri);
-        }
-        return false;
+        // we also want to send back the logged in user's uri
+        $userUri = $this->getUserUri($userId);
+
+        return array('access_token' => $accessToken, 'user_uri' => $userUri);
+    }
+
+    protected function createUserFromTrustedEmail($email, $fullName, $userName)
+    {
+        $sql = "
+            INSERT INTO user
+            SET email = :email,
+                verified = 1,
+                active = 1,
+                full_name = :fullName,
+                username = :userName
+        ";
+
+        $stmt = $this->_db->prepare($sql);
+        $stmt->execute(
+            [
+                "email" => $email,
+                'fullName' => $fullName,
+                'userName' => $userName
+            ]
+        );
+
+        return [
```

Comment:
Don't we need a failback here? What happens when executing the statement fails? Perhaps because the user already exists?

Reply:
The check for an existing user happens before this statement is run. It tries to use that user first. So probably no fall back needed as this would any fail if the SQL is invalid